### PR TITLE
implement cleanup option for rados engine

### PR DESCRIPTION
**Pull request by brettmilford**
implements a cleanup option for the rados engine.
This is to resolve issue #760

### CODE REVIEW

**Comment by sitsofe** [comment]
@brettmilford Could you squash these commits?

**Comment by sitsofe** [comment]
@brettmilford somehow a build breakage was triggered...

**Comment by brettmilford** [comment]
Ah yeah. It appears to be in an unrelated/untouched piece of code; any ideas @sitsofe?

**Comment by sitsofe** [comment]
@brettmilford Hmm, I can only see one call to it from `rbd_iter_events()`. Maybe it could be hidden inside a `#ifdef CONFIG_RBD_POLL` guard (or inlined directly at the sole call site)?

I've no idea why this wouldn't have triggered earlier though.

**Comment by sitsofe** [comment]
@brettmilford I think this one got lost... did you want to try and finish it off?

**Comment by brettmilford** [comment]
Hmm, looks like I chose the wrong commit to fork off. I think master at the time (971341ad) produces the same error - will try and rebase off a tagged commit.

**Comment by sitsofe** [comment]
@brettmilford nice! Could you fix up the commit message to be in fio's usual style and add your signed off by?

**Comment by sitsofe** [comment]
(I think that mainly means adding an rbd: prefix to the title)

**Comment by sitsofe** [comment]
Oh and you might want to include a fixes https://github.com/axboe/fio/issues/760 line...

**Comment by brettmilford** [comment]
@sitsofe `rbd:` or `rados:` - it should only change the rados engine.

**Comment by sitsofe** [comment]
Let's go with rados.

**Comment by sitsofe** [comment]
(@brettmilford Quick ping re last comment)

**Comment by brettmilford** [comment]
@sitsofe thanks, not sure why I didn't get notified.
Um - I should probably default this to True to retain the expected behavior hey?

File: HOWTO [hunk 1]
```diff
@@ -2157,6 +2157,10 @@ with the caveat that when used on the command line, they must come after the
         Poll store instead of waiting for completion. Usually this provides better
         throughput at cost of higher(up to 100%) CPU utilization.
 
+.. option:: cleanup=bool : [rados]
+
+        Clean up rados objects after run.
+
```
Comment (sitsofe) [inline HOWTO]: So just state the default and don't forget to update the fio.1 man page too...

**Comment by sitsofe** [comment]
> Um - I should probably default this to True to retain the expected behavior hey?

You're entirely correct - my bad!

**Comment by axboe** [comment]
Did this all get resolved?

**Comment by brettmilford** [comment]
Hi @sitsofe @axboe, sorry for the delayed response, I needed a chance to test this properly. This patch doesn't appear to work as expected, but also I can't seem to get the same behavior in the version compiled from source as the version packaged in fedora 21 (that is the cleanup function doesn't work at all when I compile from source). In light of this, and a lack of time to delve deeper into why this is, I'm going to close of this PR, Thanks for your assistance.

**Pull request closed without merging**